src/ucp/proto/proto_perf.c

```diff
@@ -0,0 +1,400 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "proto_perf.h"
+#include "proto_debug.h"
+
+#include <ucs/datastruct/list.h>
+#include <ucs/debug/memtrack_int.h>
+#include <ucs/sys/string.h>
+
+
+struct ucp_proto_perf_segment {
+    /* List element */
+    ucs_list_link_t       list;
+
+    /* Start value of this segment (inclusive) */
+    size_t                start;
+
+    /* End value of this segment (inclusive) */
+    size_t                end;
```

Comment:
yes, it's done in order to support holes, including starting from certain min_length and that for some ranges a protocol would not be found

Reply:
Are there at least theoretical cases for protocols with holes? 

I am asking since from my point of view, supporting holes can cause extra complication during summarization of two separate perf structures. I mean if we have two perf structures for different protocol stages and we need to sum up them, therefore they seem to need to conform each other (have same global start, end and holes). Maybe supporting global start and end would be enough? WDYT?

Reply:
After the discussion it was decided to proceed with holes support.